t/Test24-reference_fregrid.sh

```diff
@@ -24,7 +24,7 @@
 
 @test "Test fregrid ocean data" {
 
-  if [ "$CC" != "icc" ]; then skip "Test fails reference check without icc"; fi
+  if ! type icc; then skip "Test fails reference check without icc"; fi
```

Comment:
I don't think this is the solution we want.  It is possible to build w/o Intel, but have the intel module loaded (or have `icc` in the PATH.

The reason `$CC` is not set is automaker does not pass the build environment variables to the shell when running tests.  Where the change _should_ be made is to have the `Makefile.am` pass `CC` to the shell that runs the tests.  There should be something already in one of the `Makefile.am`s.

Even with that, this test will still not work as desired.  It is possible to set `CC` to the full path.  This would also fail the check.  We will also run into a time when even the Intel compiler may not bit-wise reproduce.  

I feel it is better to drop the bit-wise tests, or to add them as a special case (e.g. `make bitwise-check`).  This will leave this check up to the user -- as the user should know if bit-wise reproducibility is expected.  

Note, if we are going to continue bit-wise checking, we should get bit-wise checking for more compilers, and not just Intel.